node-wasm/src/worker/channel.rs

```diff
@@ -0,0 +1,211 @@
+use std::fmt::{self, Debug};
+
+use js_sys::JsString;
+use serde_wasm_bindgen::{from_value, to_value};
+use tokio::sync::{mpsc, Mutex};
+use tracing::{debug, error, info, warn};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::spawn_local;
+use web_sys::{MessageEvent, MessagePort, SharedWorker};
+
+use crate::utils::WorkerSelf;
+use crate::worker::commands::{NodeCommand, WorkerResponse};
+use crate::worker::WorkerError;
+
+type WireMessage = Result<WorkerResponse, WorkerError>;
+type WorkerClientConnection = (MessagePort, Closure<dyn Fn(MessageEvent)>);
+
+const WORKER_CHANNEL_SIZE: usize = 1;
```

Comment:
Add a comment here why this doesn't need to be more than one.

Reply:
Added comment 👍